src/flag_gems/utils/pointwise_dynamic.py

```diff
@@ -50,12 +51,18 @@ def __init__(
         is_tensor: Optional[List[bool]] = None,
         dtypes: Optional[List[Optional[type]]] = None,
         num_outputs: Optional[int] = None,
-        output_dtypes: Optional[List[torch.dtype]] = None,
+        promotion_methods: Optional[List[Tuple[int, ...]]] = None,
```

Comment:
What sbout still support `List[torch.dtype]`?

Reply:
I thought because `pointwise_dynamic` is for element_wise ops and once we add type promotion it can deal with more complicated input type combination，so we should use promotion to get all kind output type for different type combination. If we still use `List[torch.dtype]` to specify output，on the one hand，it‘s hard to figure out all kind output type for different type combination, on the other hand, it makes  the logic of `pointwise_dynamic` complicated and confusing